src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition } from '../types';
+import { Datatable, getType } from '../../expression_types';
+
+export interface MapColumnArguments {
+  id?: string | null;
+  name: string;
+  expression?: (datatable: Datatable) => Promise<boolean | number | string | null>;
+  copyMetaFrom?: string | null;
+}
+
+export const mapColumn: ExpressionFunctionDefinition<
+  'mapColumn',
+  Datatable,
+  MapColumnArguments,
+  Promise<Datatable>
+> = {
+  name: 'mapColumn',
+  aliases: ['mc'], // midnight commander. So many times I've launched midnight commander instead of moving a file.
+  type: 'datatable',
+  inputTypes: ['datatable'],
+  help: i18n.translate('expressions.functions.mapColumnHelpText', {
+    defaultMessage:
+      'Adds a column calculated as the result of other columns. ' +
+      'Changes are made only when you provide arguments.' +
+      'See also {alterColumnFn} and {staticColumnFn}.',
+    values: {
+      alterColumnFn: '`alterColumn`',
+      staticColumnFn: '`staticColumn`',
+    },
+  }),
+  args: {
+    id: {
+      types: ['string', 'null'],
+      help: i18n.translate('expressions.functions.mapColumn.args.idHelpText', {
+        defaultMessage:
+          'An optional id of the resulting column. When `null` the name/column argument is used as id.',
+      }),
+      required: false,
+      default: null,
+    },
+    name: {
+      types: ['string'],
+      aliases: ['_', 'column'],
+      help: i18n.translate('expressions.functions.mapColumn.args.nameHelpText', {
+        defaultMessage: 'The name of the resulting column.',
+      }),
+      required: true,
+    },
+    expression: {
+      types: ['boolean', 'number', 'string', 'null'],
+      resolve: false,
+      aliases: ['exp', 'fn', 'function'],
+      help: i18n.translate('expressions.functions.mapColumn.args.expressionHelpText', {
+        defaultMessage:
+          'A {CANVAS} expression that is passed to each row as a single row {DATATABLE}.',
```

Comment:
I know that you didn't write this text, but maybe if you're making changes you can correct it. Two things to correct:

1. The row is passed into the expression, but the text has it backwards
2. Canvas expression vs expression

```suggestion
          'An expression that is executed on every row, provided with a single-row {DATATABLE} context and returning the cell value.',
```